projects/ad355xr_iio/app/app_config.c

```diff
@@ -0,0 +1,299 @@
+	.extra = &spi_dma_tx_stop_pwm_gpio_extra_init_params
+};
+
+/* PWM init parameters for timer which will stop spi dma transfer */
+static struct no_os_pwm_init_param spi_dma_tx_stop_pwm_init_params = {
+	.id = SPI_DMA_TX_STOP_PWM_ID,
+	.period_ns = CONV_PERIOD_NSEC(MAX_SAMPLING_RATE),
+	.duty_cycle_ns = CONV_DUTY_CYCLE_NSEC(MAX_SAMPLING_RATE,LDAC_PWM_DUTY_CYCLE),
+	.platform_ops = &pwm_ops,
+	.extra = &spi_dma_tx_stop_pwm_extra_init_params,
+#if (ACTIVE_PLATFORM == STM32_PLATFORM)
+	.pwm_gpio = &spi_dma_tx_stop_pwm_gpio_init
+#endif
+};
+#endif
+
+/* External interrupt init parameters used for ldac */
+static struct no_os_irq_init_param trigger_gpio_irq_params = {
+	.irq_ctrl_id = IRQ_CTRL_ID,
+	.platform_ops = &irq_platform_ops,
+	.extra = &ext_int_extra_init_params
+};
+
+/* External interrupt callback descriptor used for ldac */
+static struct no_os_callback_desc ext_int_callback_desc = {
+	.callback = ldac_pos_edge_detect_callback,
+	.event = NO_OS_EVT_GPIO,
+	.peripheral = NO_OS_GPIO_IRQ,
+	.handle = trigger_gpio_handle
+};
+
+/* UART descriptor for IIO communication */
+struct no_os_uart_desc *uart_iio_com_desc;
+
+/* UART descriptor for console communication */
+struct no_os_uart_desc *uart_console_stdio_desc;
+
+/* PWM descriptor for ldac */
+struct no_os_pwm_desc *ldac_pwm_desc;
+
+/* PWM descriptor to stops SPI DMA transfer */
+struct no_os_pwm_desc* spi_dma_tx_stop_pwm_desc;
+
+/* External interrupt descriptor */
+struct no_os_irq_ctrl_desc *trigger_irq_desc;
+
+/******************************************************************************/
+/************************ Functions Prototypes ********************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/**
+ * @brief 	Initialize the PWM trigger contoller for ldac
+ * @return	0 in case of success, negative error code otherwise
+ */
+int32_t init_ldac_pwm_trigger(void)
+{
+	int32_t ret;
+
+	ret = no_os_pwm_init(&ldac_pwm_desc, &ldac_pwm_init_params);
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief 	Initialize the timer in pwm mode which used to stops spi dma transfer
+ * @return	0 in case of success, negative error code otherwise
+ */
+int32_t init_spi_dma_tx_stop_pwm(void)
+{
+	int32_t ret;
+
+#if (INTERFACE_MODE == SPI_DMA)
+	ret = no_os_pwm_init(&spi_dma_tx_stop_pwm_desc,
+			     &spi_dma_tx_stop_pwm_init_params);
+	if (ret) {
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+/**
+ * @brief Interrupt Service Routine to monitor ldac positive edge.
+ * @param ctx[in] - Callback context (unused)
+ * @return None
+ * @note this function is unused. we are handling ldac interrupt
+ * directly from EXTI IRQHandler.
+ */
+void ldac_pos_edge_detect_callback(void* ctx)
```

Comment:
Is this needed?

Reply:
Yes. It is used to register the callback for gpio interrupt.

Reply:
But we aren't doing anything inside it? So, what do we need the callback for?

Reply:
no_os_irq_register_callback function is needed to register and enable interrupt.